### PR TITLE
Python fairness module

### LINKED ISSUE #342: fairness module
add new fairness module - `model_fairness()`

### PR DESCRIPTION
**Pull request by jakwisn**
Adding module for fairness to python dalex. The module is based on R package `fairmodels`. PR has a backend and 2 plots present and further work will be gradually added. closes #342

### CODE REVIEW

File: python/dalex/dalex/datasets/_load.py [hunk 1]
```diff
@@ -168,3 +168,29 @@ def load_hr_test():
     dataset = pd.read_csv(abs_datasets_path, index_col=0, sep=';')
 
     return dataset
+
+
+def load_german():
```
Comment (hbaniecki) [inline python/dalex/dalex/datasets/_load.py:173]: Do we want all lowercase column names?

File: python/dalex/dalex/_explainer/object.py [hunk 2]
```diff
@@ -755,10 +758,88 @@ def model_surrogate(self,
...
+        privileged : str
+            Subgroup that is suspected to have the most privilege.
+            It needs to be a string present in `protected`.
+        cutoff : float or dict, optional
+            Threshold for probabilistic output of a classifier.
+            It might be:
+                float - same for all subgroups from `protected`
+                dict - individually adjusted for each subgroup
+                       (must have values from `protected` as keys).
+        kwargs :
```
Comment (hbaniecki) [inline python/dalex/dalex/_explainer/object.py:790]: If `verbose` is `True` by default it must be a named parameter.

**Review by hbaniecki — COMMENTED** [review]
Why not make `basics` -> `_basics` and `group_fairness` -> `_group_fairness` like the rest of the package? Now `dx.fairness.<tab>` shows unnecessary propositions.

File: python/dalex/dalex/fairness/group_fairness/object.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        df_ratios = calculate_ratio(sub_confusion_matrix_metrics, self.privileged)
+        parity_loss = calculate_parity_loss(sub_confusion_matrix_metrics, self.privileged)
+
+        self._subgroup_confusion_matrix_metrics_object = sub_confusion_matrix_metrics
+        self.subgroup_metrics = sub_confusion_matrix_metrics.to_horizontal_DataFrame()
+        self.parity_loss = parity_loss
+        self.result = df_ratios
+        self.label = label
+
+    def fairness_check(self, epsilon=0.8, verbose=True):
```
Comment (hbaniecki) [inline python/dalex/dalex/fairness/group_fairness/object.py:32]: Was wondering: `check`? `check_fairness`? usually a verb(+noun)

Comment (jakwisn) [reply python/dalex/dalex/fairness/group_fairness/object.py:32]: Yeah I was about to name it that in R package but changed order to adhere to other function names. There was `fairness_heatmap`, `fairness_pca` etc. so it was nice to have something similar.

File: python/dalex/dalex/_explainer/object.py (hunk 2 above)
Comment (jakwisn) [reply python/dalex/dalex/_explainer/object.py:790]: Changed default to `False` and added `verbose_cat`  to checks. Now hen user sets `verbose = True` he/she will see information about changing parameters. Hope this is okay

File: python/dalex/dalex/datasets/_load.py (hunk 1 above)
Comment (jakwisn) [reply python/dalex/dalex/datasets/_load.py:173]: Done

**Review by hbaniecki — APPROVED** [review]
LGTM

**Pull request merged**